apps/mobile-wallet/src/features/addressesManagement/AddressDetailsModal.tsx

```diff
@@ -61,34 +59,32 @@ const AddressDetailsModal = withModal<AddressDetailsModalProps>(({ id, addressHa
     <BottomModal modalId={id} title={<AddressBadge addressHash={addressHash} fontSize={16} />}>
       <Content>
         <RoundedCard>
-          <AnimatedBackground shade={address?.settings.color} isAnimated />
+          <AddressAnimatedBackground addressHash={addressHash} />
           <BalanceSummary addressHash={addressHash} />
         </RoundedCard>
 
         <ActionButtons>
-          {addressTokens.length > 0 && (
-            <ActionCardButton title={t('Send')} onPress={handleSendPress} iconProps={{ name: 'send' }} />
-          )}
+          {hasTokens && <ActionCardButton title={t('Send')} onPress={handleSendPress} iconProps={{ name: 'send' }} />}
           <ActionCardButton title={t('Receive')} onPress={handleReceivePress} iconProps={{ name: 'download' }} />
           <ActionCardButton title={t('Settings')} onPress={handleSettingsPress} iconProps={{ name: 'settings' }} />
-          <SetDefaultAddressCardButton address={address} />
+          <SetDefaultAddressCardButton addressHash={addressHash} />
```

Comment:
and same for this component now. Any changes in the `Address` object that it makes will not affect the parent component.